src/main/java/io/cdap/plugin/http/source/batch/HttpInputFormatProvider.java

```diff
@@ -47,4 +66,44 @@ public String getInputFormatClassName() {
   public Map<String, String> getInputFormatConfiguration() {
     return conf;
   }
+  @Nullable
+  public Schema getSchema(FormatContext context) {
+    PageFormat format = config.getFormat();
+    if (format != PageFormat.CSV && format != PageFormat.TSV) {
+      // if the format is not CSV or TSV, then we don't infer the schema
+      return null;
+    }
+    FailureCollector failureCollector = context.getFailureCollector();
+    HttpPartialContentReader partialContentReader = new HttpPartialContentReader(config);
```

Comment:
this is csv specific code, move it to another file, so in future if we need to add for other format types those can be accomodated without changing this class. You can also use an interface in between

Reply:
Added the logic to a new class `DelimitedSchemaDetector`